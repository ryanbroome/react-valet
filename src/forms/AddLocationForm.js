import React, { useState, useContext } from "react";
import { useHistory } from "react-router-dom";
import { FormGroup, Form, Input, Label, Button, FormText } from "reactstrap";
import UserContext from "../auth/UserContext";

const AddLocationForm = () => {
  const history = useHistory();
  const { addLocation } = useContext(UserContext);

  const INITIAL_STATE = {
    sitename: "",
  };

  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    addLocation(formData.sitename);
    setFormData(INITIAL_STATE);
    history.push(`/`);
  };

  return (
    <Form onSubmit={handleSubmit}>
      <h3>New Location Form</h3>
      <Button color="primary">Add Location</Button>
      {Object.keys(INITIAL_STATE).map((val, idx) => (
        <FormGroup key={`FormGroup-${val}`}>
          <Label
            htmlFor={val}
            key={`Label-${idx}`}></Label>
          <Input
            id={val}
            key={`Input-${val}`}
            type="text"
            name={val}
            placeholder={val.toLowerCase()}
            value={formData[val]}
            onChange={handleChange}></Input>
        </FormGroup>
      ))}
    </Form>
  );
};

export default AddLocationForm;
