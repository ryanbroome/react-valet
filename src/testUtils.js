import React from "react";
import UserContext from "./auth/UserContext";
// import { render } from "@testing-library/react";
// import { MemoryRouter } from "react-router";
import App from "../src/App";
// import TransactionList from "./TransactionList";

const MockUserContext = ({ children }) => {
  const mockToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3R1c2VyIiwibG9jYXRpb25JZCI6MSwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTcxMzA3NTM4Mn0.dXlnawfzk6wlsgpouoSp58QLvlOOSJFC03iq_FXab-I";
  const mockUserDetail = {
    userId: 1,
    vehicleId: 1,
    locationId: 1,
    transactionId: 1,
    ticketNum: "1",
    mobile: "5553334444",
    color: "red",
    make: "honda",
    username: "testuser",
  };
  const mockLocation = {};
  const mockVehicle = {};
  const mockTransaction = {};

  return <UserContext.Provider value={{ token: mockToken, userDetail: mockUserDetail, location: mockLocation, vehicle: mockVehicle, transaction: mockTransaction, update: App.update }}>{children}</UserContext.Provider>;
};

export { MockUserContext };
