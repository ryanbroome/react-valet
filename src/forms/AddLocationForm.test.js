import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import AddLocationForm from "./AddLocationForm";
import { MemoryRouter } from "react-router";
import { MockUserContext } from "../testUtils";

it("AddLocationForm: matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MockUserContext children={<AddLocationForm />}></MockUserContext>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("AddLocationForm: renders without crashing", function () {
  render(
    <MemoryRouter>
      <MockUserContext children={<AddLocationForm />}></MockUserContext>
    </MemoryRouter>
  );
});
