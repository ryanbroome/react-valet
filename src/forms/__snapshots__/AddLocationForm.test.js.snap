// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddLocationForm: matches snapshot 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <h3>
      New Location Form
    </h3>
    <button
      class="btn btn-primary"
    >
      Add Location
    </button>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="sitename"
      />
      <input
        class="form-control"
        id="sitename"
        name="sitename"
        placeholder="sitename"
        type="text"
        value=""
      />
    </div>
  </form>
</DocumentFragment>
`;
